server/src/main/java/org/apache/cassandra/sidecar/handlers/livemigration/LiveMigrationListInstanceFilesHandler.java

```diff
@@ -0,0 +1,99 @@
+import org.apache.cassandra.sidecar.acl.authorization.BasicPermissions;
+import org.apache.cassandra.sidecar.cluster.InstancesMetadata;
+import org.apache.cassandra.sidecar.cluster.instance.InstanceMetadata;
+import org.apache.cassandra.sidecar.common.response.InstanceFilesListResponse;
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+import org.apache.cassandra.sidecar.config.SidecarConfiguration;
+import org.apache.cassandra.sidecar.handlers.AbstractHandler;
+import org.apache.cassandra.sidecar.handlers.AccessProtected;
+import org.apache.cassandra.sidecar.livemigration.CassandraInstanceFiles;
+import org.apache.cassandra.sidecar.livemigration.CassandraInstanceFilesImpl;
+import org.apache.cassandra.sidecar.utils.CassandraInputValidator;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+import org.jetbrains.annotations.NotNull;
+
+import static org.apache.cassandra.sidecar.utils.HttpExceptions.wrapHttpException;
+
+/**
+ * Endpoint for returning list of files that a destination instance can copy.
+ */
+public class LiveMigrationListInstanceFilesHandler extends AbstractHandler<Void> implements AccessProtected
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LiveMigrationListInstanceFilesHandler.class);
+    private final SidecarConfiguration sidecarConfiguration;
+    private final InstancesMetadata instancesMetadata;
+
+    @Inject
+    public LiveMigrationListInstanceFilesHandler(InstanceMetadataFetcher metadataFetcher,
+                                                 ExecutorPools executorPools,
+                                                 CassandraInputValidator validator,
+                                                 SidecarConfiguration sidecarConfiguration,
+                                                 InstancesMetadata instancesMetadata)
+    {
+        super(metadataFetcher, executorPools, validator);
+        this.sidecarConfiguration = sidecarConfiguration;
+        this.instancesMetadata = instancesMetadata;
+    }
+
+    protected Void extractParamsOrThrow(RoutingContext context)
+    {
+        return null;
+    }
+
+    protected void handleInternal(RoutingContext context, HttpServerRequest httpRequest, @NotNull String host,
+                                  SocketAddress remoteAddress, Void request)
+    {
+        InstanceMetadata instanceMetadata = instancesMetadata.instanceFromHost(host);
+
+        CassandraInstanceFiles filesList = new CassandraInstanceFilesImpl(instanceMetadata,
+                                                                          sidecarConfiguration.liveMigrationConfiguration());
+        try
+        {
+
+            context.json(new InstanceFilesListResponse(filesList.getFiles()));
+        }
+        catch (IOException e)
+        {
+            LOGGER.error("Could not fetch instance files information.", e);
+            context.fail(wrapHttpException(HttpResponseStatus.INTERNAL_SERVER_ERROR, e.getMessage(), e));
+        }
```

Comment:
This needs to run in a blocking service thread
```suggestion
                executorPools.service().runBlocking(() -> {
            try
            {
                context.json(new InstanceFilesListResponse(filesList.getFiles()));
            }
            catch (IOException e)
            {
                LOGGER.error("Could not fetch instance files information.", e);
                context.fail(wrapHttpException(HttpResponseStatus.INTERNAL_SERVER_ERROR, e.getMessage(), e));
            }
        });
```

Reply:
done